sklearn/cluster/tests/test_k_means.py

```diff
@@ -987,3 +988,14 @@ def test_iter_attribute():
     estimator = KMeans(algorithm="elkan", max_iter=1)
     estimator.fit(np.random.rand(10, 10))
     assert estimator.n_iter_ == 1
+
+
+def test_change_n_init_future_warning():
```

Comment:
I did not think about it at first.
Can you check `kmeans` function as well.
Using only the class will not check for the raising in the function itself since `self.n_init` will be set in the meanwhile.